pkg/backoff/backoff_internal_test.go

```diff
@@ -0,0 +1,85 @@
+package backoff
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/wakatime/wakatime-cli/pkg/config"
+
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUpdateBackoffSettings(t *testing.T) {
+	v := viper.New()
+
+	tmpFile, err := ioutil.TempFile(os.TempDir(), "wakatime")
+	require.NoError(t, err)
+
+	defer os.Remove(tmpFile.Name())
+
+	v.Set("config", tmpFile.Name())
+
+	err = updateBackoffSettings(v, 2, nil)
+	require.NoError(t, err)
+
+	ini, err := config.NewIniWriter(v, func(vp *viper.Viper) (string, error) {
+		assert.Equal(t, v, vp)
+		return tmpFile.Name(), nil
+	})
+	require.NoError(t, err)
+
+	assert.NotEmpty(t, ini.File.Section("internal").Key("backoff_at").String())
```

Comment:
Let's assert the actual value with `assert.WithinDuration` (https://pkg.go.dev/github.com/stretchr/testify/assert#WithinDuration). As you cannot predict the now time, `WithinDuration` will allow you to ensure that the new config value is a certain timeframe (e.g. now +/- one second)

Reply:
Ohh didn't know about that function. Good to know, thanks.